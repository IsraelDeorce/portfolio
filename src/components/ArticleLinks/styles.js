import styled from 'styled-components'

export const ArticleLinks = styled.ul`
  display: flex;
  flex-direction: column;
  padding-left: 1em;
  & > * {
    color: ${({ theme }) => theme.c_brackets};
    &:hover {
      color: ${({ theme }) => theme.c_varType};
    }
    padding-bottom: 1em;
  }
`
