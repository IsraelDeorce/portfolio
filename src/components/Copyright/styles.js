import styled from 'styled-components'

export const Copyright = styled.div`
  display: flex;
  justify-content: center;
`
