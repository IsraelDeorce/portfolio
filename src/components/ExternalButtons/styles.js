import styled from 'styled-components'

export const ExternalButtons = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 20px;
  font-size: 1.25rem;
`
