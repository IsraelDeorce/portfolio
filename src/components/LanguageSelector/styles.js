import styled from 'styled-components'

export const LanguageSelector = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
  color: ${({ theme }) => theme.c_brackets};
`

export const LanguageCode = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`