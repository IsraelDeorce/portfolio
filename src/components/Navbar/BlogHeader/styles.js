import styled from 'styled-components'

export const BlogHeader = styled.nav`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  font-weight: 700;
  background-color: ${({ theme }) => theme.c_menu};
  color: ${({ theme }) => theme.c_brackets};
  padding: 0 1.5em;
  min-height: 60px;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
  @media (min-width: 992px) {
    display: none;
  }
`
export const Hamburguer = styled.span`
  display: flex;
  align-items: center;
  cursor: pointer;
  flex-shrink: 0;
  &:hover {
    color: ${({ theme }) => theme.c_varType};
  }
`
