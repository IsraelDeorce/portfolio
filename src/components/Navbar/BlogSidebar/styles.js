import styled from 'styled-components'

export const Background = styled.div`
  opacity: 0.7;
  background-color: ${({ theme }) => theme.c_background};
  position:fixed;
  width:100%;
  height:100%;
  top:0px;
  left:0px;
  @media (min-width: 992px) {
    display: none;
  }
`

export const BlogSidebar = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.c_menu};
  height: 100vh;
  width: 65vw;
  z-index: 2;
  box-shadow: 0 5px 30px ${({ theme }) => theme.c_tab};
  padding: 0 1em 1em 1em;
  & > * {
    width: 100%;
  }
  & > :last-child {
    align-self: center;
  }
  @media (min-width: 992px) {
    position: static;
    top: auto;
    left: auto;
    right: auto;
    height: auto;
    width: 285px;
    min-width: 285px;
    box-shadow: none;
  }
`

export const Header = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  min-height: 60px;
  border-bottom: 2px solid gray;
  & > *:not(:last-child) {
    padding: 0.35em;
  }
`

export const Preferences = styled.div`
  display: flex;
  align-items: center;
  & > *:not(:last-child) {
    margin-right: 0.75em;
  }
`
