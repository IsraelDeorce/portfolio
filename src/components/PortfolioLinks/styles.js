import styled, { css } from 'styled-components'

export const PortfolioLinks = styled.div`
  display: flex;
  color: ${({ theme }) => theme.c_brackets};
  & > *:hover {
    color: ${({ theme }) => theme.c_varType};
  }
  ${({ orientation }) => orientation === 'horizontal' && css`
    & > *:not(:last-child) {
      margin-right: 2em;
    }
  `}
  ${({ orientation }) => orientation === 'vertical' && css`
    flex-direction: column;  
    & > * {
      margin: .75em 0;
    }
  `}
`
