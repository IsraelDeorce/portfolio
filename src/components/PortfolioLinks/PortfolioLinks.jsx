import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import * as S from './styles'

const propTypes = {
  orientation: PropTypes.oneOf(['horizontal', 'vertical']),
  onCLick: PropTypes.func
}

function PortfolioLinks({ orientation = 'horizontal', onClick }) {
  const { t } = useTranslation()

  return (
    <S.PortfolioLinks orientation={orientation} onClick={onClick}>
      <a href='#skills'>{t('button.skills')}</a>
      <a href='#experience'>{t('button.experience')}</a>
      <a href='#sec3'>{t('button.education')}</a>
      <a href='#sec4'>{t('button.contact')}</a>
    </S.PortfolioLinks>
  )
}

PortfolioLinks.propTypes = propTypes

export default PortfolioLinks
