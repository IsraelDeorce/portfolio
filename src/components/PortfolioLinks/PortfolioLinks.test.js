import { t } from 'react-i18next'
import PortfolioLinks from './PortfolioLinks'
import * as S from './styles' 

describe('<PortfolioLinks/>', () => {
  afterEach(() => jest.clearAllMocks())

  it('renders four portfolio links', () => {
    const shallowed = shallow(<PortfolioLinks />).find(S.PortfolioLinks)
    expect(shallowed.children()).toHaveLength(4)
  })

  it('renders with horizontal orientation by default', () => {
    expect(shallow(<PortfolioLinks />).find(S.PortfolioLinks).prop('orientation')).toBe('horizontal')
  })

  it('renders with vertical orientation when passed', () => {
    expect(shallow(<PortfolioLinks orientation='vertical' />).find(S.PortfolioLinks).prop('orientation')).toBe('vertical')
  })

  it('renders a link that refers to #skills with correct translation', () => {
    expect(shallow(<PortfolioLinks />).find('a[href="#skills"]').text()).toBe('button.skills')
    expect(t).toHaveBeenCalledWith('button.skills')
  })

  it('renders a link that refers to #experience with correct translation', () => {
    expect(shallow(<PortfolioLinks />).find('a[href="#experience"]').text()).toBe('button.experience')
    expect(t).toHaveBeenCalledWith('button.experience')
  })

  it('renders a link that refers to #sec3 with correct translation', () => {
    expect(shallow(<PortfolioLinks />).find('a[href="#sec3"]').text()).toBe('button.education')
    expect(t).toHaveBeenCalledWith('button.education')
  })

  it('renders a link that refers to #sec4 with correct translation', () => {
    expect(shallow(<PortfolioLinks />).find('a[href="#sec4"]').text()).toBe('button.contact')
    expect(t).toHaveBeenCalledWith('button.contact')
  })

  describe('Snapshot', () => {
    it('renders without crashing using shallow', () => {
      expect(shallow(<PortfolioLinks />)).toMatchSnapshot()
    })

    it('renders without crashing using render', () => {
      expect(render(<PortfolioLinks />)).toMatchSnapshot()
    })

    it('renders without crashing using mount', () => {
      expect(mount(<PortfolioLinks />)).toMatchSnapshot()
    })
  })
})
