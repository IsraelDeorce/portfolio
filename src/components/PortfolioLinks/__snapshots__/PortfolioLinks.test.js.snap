// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortfolioLinks/> Snapshot renders without crashing using shallow 1`] = `
<styled.div
  orientation="horizontal"
>
  <HashLink
    to="#about"
  >
    button.about
  </HashLink>
  <HashLink
    to="#skills"
  >
    button.skills
  </HashLink>
  <HashLink
    to="#experience"
  >
    button.experience
  </HashLink>
  <HashLink
    to="#contact"
  >
    button.contact
  </HashLink>
</styled.div>
`;
